decomp2d/decomp_2d.f90

```diff
@@ -485,7 +488,19 @@ end subroutine decomp_2d_init
   subroutine decomp_2d_finalize
     
     implicit none
-    
+ 
+    integer :: ierror
+
+    call MPI_COMM_FREE(DECOMP_2D_COMM_ROW, ierror)
+    if (ierror /= 0) call decomp_2d_warning(__FILE__, __LINE__, ierror, "MPI_COMM_FREE")
```

Comment:
It's good to check these - I will have a go at a macro to clean it up slightly (hopefully people will do this if it's easier to write)